src/sdk/endpoint_stats_collector.js

```diff
@@ -54,44 +58,49 @@ class EndpointStatsCollector {
         this.bucket_counters = {};
     }
 
-    async _send_stats() {
-        await P.all([this._send_endpoint_stats(), this._send_nsfs_stats()]);
+    reset_all_nsfs_stats() {
+        this.nsfs_io_counters = this._new_namespace_stats();
+        this.op_stats = {};
+        this.fs_workers_stats = {};
     }
 
     async _send_endpoint_stats() {
         await P.delay_unblocking(SEND_STATS_DELAY);
-        // clear this.send_stats to allow new updates to trigger another _send_stats
+        // clear this.send_stats to allow new updates to trigger another _send_endpoint_stats
         this.send_stats = null;
         try {
             await this.rpc_client.object.update_endpoint_stats(this.get_all_stats(), {
                 timeout: SEND_STATS_TIMEOUT
             });
             this.reset_all_stats();
         } catch (err) {
-            // if update fails trigger _send_stats again
-            dbg.error('failed on update_endpoint_stats. trigger_send_stats again', err);
+            // if update fails trigger _send_endpoint_stats again
+            dbg.error('failed on update_endpoint_stats. trigger _send_endpoint_stats again', err);
             this._trigger_send_stats();
         }
     }
 
-    // _send_nsfs_stats will not retry update_nsfs_stats, We can send it in the next iteration.
     async _send_nsfs_stats() {
+        await P.delay_unblocking(SEND_NSFS_STATS_DELAY);
         const _nsfs_stats = {
             nsfs_stats: {
                 io_stats: this.nsfs_io_counters,
                 op_stats: this.op_stats,
                 fs_workers_stats: this.fs_workers_stats,
             }
         };
-        try {
-            await this.rpc_client.stats.update_nsfs_stats(_nsfs_stats, {
-                timeout: SEND_STATS_TIMEOUT
-            });
-            this.nsfs_io_counters = this._new_namespace_stats();
-            this.op_stats = {};
-            this.fs_workers_stats = {};
-        } catch (err) {
-            dbg.error('failed on update_nsfs_stats.', err);
+        let retries = 0;
+        while (retries < SEND_STATS_MAX_RETRIES) {
```

Comment:
If I'm picky here - you only get `SEND_STATS_MAX_RETRIES-1` retries with this while loop :) 
so currently it's one attempt and 2 additional retries

Reply:
Yes, That's true.
A probably better name will be `SEND_STATS_MAX_TRIES`
Or Change it to `<=` 😄